### ISSUE TITLE
Sketch in state "Solver failed to converge" should turn sketch geometry in red

### ISSUE THREAD

**Issue by chrisb-github** [FreeCAD/FreeCAD#13880]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

Other error messages of the solver change the sketch elements color to red. This very severe one should do the same.
![Snip macro screenshot-6a558f](https://github.com/FreeCAD/FreeCAD/assets/11750017/315e2795-84e1-4135-a063-0c0737993ece)

### Full version info

```shell
OS: macOS 13.0
Word size of FreeCAD: 64-bit
Version: 0.22.0dev.36904 (Git)
Build type: Release
Branch: main
Hash: 0622cb784e199053656b50854e92eff806130408
Python 3.11.8, Qt 5.15.13, Coin 4.0.2, Vtk 9.2.6, OCC 7.7.2
Locale: C/Default (C)
```

### Subproject(s) affected?

Sketcher

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by mastercaution** [comment]
I disagree. I'm currently in a situation where I have a perfectly valid and solvable sketch, but the solver failed to converge. So, the problem seems to be in the solver, not in the sketch. Turning the sketch elements to red, would suggest that the sketch is wrong, invalid or not solvable, which would be not true in my case.
